lib/chef/provider/mount/linux.rb

```diff
@@ -32,6 +32,7 @@ class Linux < Chef::Provider::Mount::Mount
 
         def mounted?
           mounted = false
+          loop_mount_points = shell_out!("losetup --list").stdout
```

Comment:
Won't this run every time even if we don't need the data now?

Reply:
That's right, earlier suggestion was to initialize this list of mount points as soon as the `Mount` class is initialised but there the specs fail.. Since rspec's mock up functionality is not ready by the time `Mount` class is initialized.  By running it once here we make sure its not run in the loop while we are going through all the mounts points in system with command `findmnt -rn`.

Reply:
Also at this point of time in code its not possible for us if the mount is of type loop or not for sure . We need this data afterwards to make sure we have information of `loop` mount points

Reply:
@tas50 does the PR look good ? or any need of updates ?

Reply:
I still don't understand why we can't just do this shellout when we need the data vs. frontloading the work and doing extra shellouts we might not need.

Reply:
@tas50 because calling losetup 93048234092834 times is bad. I asked for that.